pkg/apis/kops/validation/validation.go

```diff
@@ -599,9 +603,75 @@ func validateNetworkingCalico(v *kops.CalicoNetworkingSpec, e *kops.EtcdClusterS
 		allErrs = append(allErrs, IsValidValue(fldPath.Child("iptablesBackend"), &v.IptablesBackend, valid)...)
 	}
 
+	if v.IPv4AutoDetectionMethod != "" {
+		allErrs = append(allErrs, validateCalicoAutoDetectionMethod(fldPath.Child("ipv4AutoDetectionMethod"), &v.IPv4AutoDetectionMethod, ipv4.Version)...)
+	}
+
+	if v.IPv6AutoDetectionMethod != "" {
+		allErrs = append(allErrs, validateCalicoAutoDetectionMethod(fldPath.Child("ipv6AutoDetectionMethod"), &v.IPv6AutoDetectionMethod, ipv6.Version)...)
+	}
+
 	return allErrs
 }
 
+func validateCalicoAutoDetectionMethod(fldPath *field.Path, runtime *string, version int) field.ErrorList {
+	methodFirstFound := "first-found"
+	methodCanReach := "can-reach="
+	methodInterface := "interface="
+	methodSkipInterface := "skip-interface="
+	validationError := field.ErrorList{}
+
+	// validation code is based on the checks in calico/node startup code
+	// valid formats are "first-found", "can-reach=DEST", or
+	// "(skip-)interface=<COMMA-SEPARATED LIST OF INTERFACES>"
+	//
+	// We won't do deep validation of the values in this check, since they can
+	// be actual interface names or regexes
+	if *runtime == methodFirstFound {
+		return field.ErrorList{}
+
+	} else if strings.HasPrefix(*runtime, methodCanReach) {
+		destStr := strings.TrimPrefix(*runtime, methodCanReach)
+		if version == ipv4.Version {
```

Comment:
Perhaps use a `switch` statement? Or pass a `func (*field.Path, string) field.ErrorList` instead of a `string`.

Reply:
I hope this is more readable!